airbyte-config/init/src/main/java/io/airbyte/config/init/JsonDefinitionsHelper.java

```diff
@@ -7,9 +7,28 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.node.BooleanNode;
 import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.StandardDestinationDefinition;
+import io.airbyte.config.StandardSourceDefinition;
 
 public class JsonDefinitionsHelper {
 
+  public static StandardSourceDefinition patchSourceDefinition(final StandardSourceDefinition sourceDefinition) {
+    final JsonNode jsonNode = Jsons.jsonNode(sourceDefinition);
+    addMissingTombstoneField(jsonNode);
+    addMissingPublicField(jsonNode);
+    addMissingCustomField(jsonNode);
+    return Jsons.object(jsonNode, StandardSourceDefinition.class);
+  }
```

Comment:
The definition models have setters ( eg: `setTombstone`, I find it awkward to require an intermediate cast to JsonNode to add these fields.

Reply:
moving to just setting a default on the standardsourcedefinition and standarddestinationdefinition as this is unexpected behavior to me and all this stuff shouldn't be necessary imo